src/utils/protocol/importProtocol.js

```diff
@@ -86,6 +77,28 @@ const extractZip = inEnvironment((environment) => {
   throw new Error(`extractZip() not available on platform ${environment}`);
 });
 
+const loadZip = inEnvironment((environment) => {
+  if (environment === environments.CORDOVA || environment === environments.ELECTRON) {
+    return source =>
+      readFile(source)
+        .then(data => Zip.loadAsync(data))
+        .catch(friendlyErrorMessage("We couldn't find a Network Canvas protocol at the location you gave us. Check the location, and try again."));
+  }
+
+  throw new Error(`loadZip() not available on platform ${environment}`);
```

Comment:
Do `loadZip()` errors not trigger a friendly error? What happens to them? What should the user do?

Reply:
There are two workflows going on here:

1. You called a method that isn't supported on the platform
1. One of the steps threw an error.

This is the first one, which will probably go straight to the console at the moment, since it's the wrong type for our promise chains (separate issue: it's generally good practice to always return the same type, so maybe we should return a promise rejection here?).

Should we capture this kind of error for the user? I'm not sure. What do you thing?

Reply:
I'm just thinking in terms of testing. Users won't have the console open, so will have no means of knowing what happened. I realise this is basically a noop, but we want to show an error when things that shouldn't happen do. I think returning a promise rejection is the right thing to do.